filters/apiusagemonitoring/filter.go

```diff
@@ -133,19 +134,22 @@ func (f *apiUsageMonitoringFilter) getRealmClientKey(r *http.Request, path *path
 	}
 
 	// if client does not match ==> realm.{no-match}
-	if !path.ClientTracking.ClientTrackingMatcher.MatchString(client) {
+	matcher := path.ClientTracking.ClientTrackingMatcher
+	if matcher == nil || !matcher.MatchString(client) {
 		return realm + "." + noMatchPlaceholder
 	}
 
 	// all matched ==> realm.client
 	return realm + "." + client
 }
 
-// resolvePath tries to match the request's path with one of the configured path template.
-func (f *apiUsageMonitoringFilter) resolvePath(req *http.Request) *pathInfo {
-	for _, p := range f.Paths {
-		if p.Matcher.MatchString(req.URL.Path) {
-			return p
+// resolveMatchedPath tries to match the request's path with one of the configured path template.
+func (f *apiUsageMonitoringFilter) resolveMatchedPath(url *url.URL) *pathInfo {
```

Comment:
the var name `url` shadows the url stdlib package. Can you please change the var to `u` for example?